src/graph/entities/attribute_set.c

```diff
@@ -212,10 +211,11 @@ void AttributeSet_Set_Allow_Null
 		return;
 	}
 
+	size_t n = ATTRIBUTESET_BYTE_SIZE(_set) + sizeof(Attribute);
+	_set = rm_realloc(_set, n);
+
 	// allocate room for new attribute
 	_set->attr_count++;
-	size_t n = ATTRIBUTESET_BYTE_SIZE(_set);
-	_set = rm_realloc(_set, n);
```

Comment:
Why was this 2 lines moved?

Reply:
@swilly22  Cause _set can be NULL, and doing _set->attr_count is invalid in this case, need first to realloc it.